hikey960.mk

```diff
@@ -0,0 +1,444 @@
+ifeq ($(DEBUG),1)
+EDK2_BUILD			?= DEBUG
+else
+EDK2_BUILD			?= RELEASE
+endif
+EDK2_BIN			?= $(EDK2_PATH)/Build/HiKey960/$(EDK2_BUILD)_$(EDK2_TOOLCHAIN)/FV/BL33_AP_UEFI.fd
+OPENPLATPKG_PATH		?= $(ROOT)/OpenPlatformPkg
+
+OUT_PATH			?=$(ROOT)/out
+MCUIMAGE_BIN			?= $(OPENPLATPKG_PATH)/Platforms/Hisilicon/HiKey960/Binary/lpm3.img
+BOOT_IMG			?=$(ROOT)/out/boot-fat.uefi.img
+GRUB_PATH			?=$(ROOT)/grub
+LLOADER_PATH			?=$(ROOT)/l-loader
+IMAGE_TOOLS_PATH		?=$(ROOT)/tools-images-hikey960
+IMAGE_TOOLS_CONFIG		?=$(OUT_PATH)/config
+PATCHES_PATH			?=$(ROOT)/patches_hikey
+STRACE_PATH			?=$(ROOT)/strace
+
+################################################################################
+# Targets
+################################################################################
+.PHONY: all
+all: arm-tf boot-img lloader strace
+
+.PHONY: clean
+clean: arm-tf-clean busybox-clean edk2-clean linux-clean optee-os-clean optee-client-clean xtest-clean helloworld-clean strace-clean update_rootfs-clean boot-img-clean lloader-clean grub-clean
+
+.PHONY: cleaner
+cleaner: clean prepare-cleaner busybox-cleaner linux-cleaner strace-cleaner grub-cleaner
+
+-include toolchain.mk
+
+.PHONY: prepare
+prepare:
+	mkdir -p $(OUT_PATH)
+
+.PHONY: prepare-cleaner
+prepare-cleaner:
+	rm -rf $(ROOT)/out
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+ARM_TF_EXPORTS ?= \
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+ARM_TF_FLAGS ?= \
+	BL32=$(OPTEE_OS_PATH)/out/arm/core/tee-pager.bin \
+	BL33=$(EDK2_BIN) \
+	SCP_BL2=$(MCUIMAGE_BIN) \
+	DEBUG=$(DEBUG) \
+	PLAT=hikey960 \
+	SPD=opteed
+
+ifeq ($(CFG_CONSOLE_UART),5)
+	ARM_TF_FLAGS += CRASH_CONSOLE_BASE=PL011_UART5_BASE
+endif
+
+.PHONY: arm-tf
+arm-tf: optee-os edk2
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
+
+.PHONY: arm-tf-clean
+arm-tf-clean:
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
+
+################################################################################
+# Busybox
+################################################################################
+BUSYBOX_COMMON_TARGET = hikey960 nocpio
+BUSYBOX_CLEAN_COMMON_TARGET = hikey960 clean
+
+.PHONY: busybox
+busybox: busybox-common
+
+.PHONY: busybox-clean
+busybox-clean: busybox-clean-common
+
+.PHONY: busybox-cleaner
+busybox-cleaner: busybox-clean-common busybox-cleaner-common
+
+################################################################################
+# EDK2 / Tianocore
+################################################################################
+EDK2_ARCH ?= AARCH64
+EDK2_DSC ?= OpenPlatformPkg/Platforms/Hisilicon/HiKey960/HiKey960.dsc
+EDK2_TOOLCHAIN ?= GCC5
+
+ifeq ($(CFG_CONSOLE_UART),5)
+	EDK2_BUILDFLAGS += -DSERIAL_BASE=0xFDF05000
+endif
+
+define edk2-call
+	$(EDK2_TOOLCHAIN)_$(EDK2_ARCH)_PREFIX=$(GCC5_AARCH64_CROSS_COMPILE) \
```

Comment:
Agree. My plan is to transition hikey.mk and hikey_debian.mk to this toolchain as well and remove legacy, but fvp and qemu v8 still use it. @jenswi-linaro Do you know if both these can also transition to gcc5? rpi3 also use it for u-boot. @igoropaniuk Same question?